src/ddscxx/include/org/eclipse/cyclonedds/core/cdr/basic_cdr_ser.hpp

```diff
@@ -182,10 +356,32 @@ void max(basic_cdr_stream& str, const T& max_sz)
  * @param[out] toread The variable to read into.
  */
 template<typename T, std::enable_if_t<std::is_enum<T>::value && !std::is_arithmetic<T>::value, bool> = true >
-void read(basic_cdr_stream& str, T& toread) {
+inline void read(basic_cdr_stream& str, T& toread) {
   read(str, *reinterpret_cast<uint32_t*>(&toread));
 }
```

Comment:
I think we'll have to look into this one too eventually as the size of enums can be changed in IDL4? We can probably get away with doing a `sizeof` of `T` and call the proper read function for the primitive.